pages/index.tsx

```diff
@@ -26,23 +27,42 @@ export default function Home({
   const [receiverError, setReceiverError] = useState(false);
   const [newReceiver, setNewReceiver] = useState('');
   const [newGiftName, setNewGiftName] = useState('');
-  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
-  const [deleteModalGiftData, setDeleteModalGiftData] = useState<Gift>();
-  const [isEditModalOpen, setIsEditModalOpen] = useState(false);
-  const [editModalGiftData, setEditModalGiftData] = useState<Gift>();
+
   const [showUserWindow, setShowUserWindow] = useState(false);
 
+  const giftQuery = useQuery({
+    queryKey: ['loadingGifts'],
+    queryFn: async () => await fetchGifts(),
+    refetchOnWindowFocus: false,
+    refetchOnMount: false,
+    retry: false,
+  });
+
+  async function fetchGifts() {
+    try {
+      const gifts = await getAllGifts();
+      setGiftData(gifts);
+      return gifts;
+    } catch (e) {
+      handleErrorToast(handleError(e));
+      throw new Error('Palvelinvirhe! Lisätietoja kehittäjäkonsolissa');
+    }
+  }
+
```

Comment:
Tämä pitäisi olla ok, `useQuery`n pitäisi osata toimia aikalailla samanlailla kuin `useEffect`in, eli odottaa ensimmäistä rendausta (voin olla aivan väärässäkin :smile: )

Otin `refetch / retry` ominaisuudet veks, koska tämä halutaan toteuttaa ainakin tähän hätään vain kerran.

Jos tuo haku halutaan tehdä useamman kerran sitten voidaan muokata noita. Tai vaihtoehtoisesti tehdä `queryKey`stä `useState`, jota muokkaamalla pitäisi lahjat `fetchGifts`in noutaa lahjat uudelleen